### PR TITLE
gossip support for new headers topic

### PR DESCRIPTION
**Pull request by gadcl**

### CODE REVIEW

File: services/gossip/topic_headers_sync.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+	for _, l := range s.handlers.headerSyncHandlers {
+		_, err := l.HandleHeaderSyncRequest(ctx, &gossiptopics.HeaderSyncRequestInput{Message: message})
+		if err != nil {
+			s.logger.Info("HandleHeaderSyncRequest failed", log.Error(err))
+		}
+	}
+}
+
+//func IsChunkTooBigError(err error) bool {
```
Comment (netoneko) [inline services/gossip/topic_headers_sync.go:164]: Commented-out code should be removed.

File: test/protocol/TransactionReceiptProof.json [hunk 2]
```diff
@@ -18,41 +18,41 @@
     "Timestamp": "15527734",
     "ReceiptsRootHash": "b3bed1cbeb01d5f991eda2a5a1dd19eef45dde3f3f3e525eb08b17cf1155503f"
   },
-  "RawResultsBlockHeader": "010000002a0000002a20000000000000200000004a64a107f0cb32536e5bce6c98c393db21cca7f4ea187ba8c4dca8b51d4ea80a36efec000000000020000000b3bed1cbeb01d5f991eda2a5a1dd19eef45dde3f3f3e525eb08b17cf1155503f200000004e07408562bedb8b60ce05c1decfe3ad16b72230967de01f640b7e4729b49fce200000003f39d5c348e5b79d06e842c114e6cc571583bbf44e4b0ebfda1a01ec05745d4320000000a25513c7e0f6eaa80a3337ee18081b9e2ed09e00af8531c8f7bb2542764027e70c0000002200000000000000",
+  "RawResultsBlockHeader": "010000002a0000002a20000000000000200000004a64a107f0cb32536e5bce6c98c393db21cca7f4ea187ba8c4dca8b51d4ea80a36efec000000000020000000b3bed1cbeb01d5f991eda2a5a1dd19eef45dde3f3f3e525eb08b17cf1155503f200000004e07408562bedb8b60ce05c1decfe3ad16b72230967de01f640b7e4729b49fce200000003f39d5c348e5b79d06e842c114e6cc571583bbf44e4b0ebfda1a01ec05745d4320000000a25513c7e0f6eaa80a3337ee18081b9e2ed09e00af8531c8f7bb2542764027e70c000000220000000000000000000000",
```
Comment (netoneko) [inline test/protocol/TransactionReceiptProof.json:21]: Why did this change?

Comment (noambergIL) [reply test/protocol/TransactionReceiptProof.json:21]: it always changes :)

File: services/gossip/codec/topic_header_sync.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		return nil, errors.New("missing SignedBatchRange")
+	}
+	if message.Sender == nil {
+		return nil, errors.New("missing Sender")
+	}
+	return [][]byte{header.Raw(), message.SignedBatchRange.Raw(), message.Sender.Raw()}, nil
+}
+
+func DecodeHeaderAvailabilityRequest(payloads [][]byte) (*gossipmessages.HeaderAvailabilityRequestMessage, error) {
+	if len(payloads) != 2 {
```
Comment (noambergIL) [inline services/gossip/codec/topic_header_sync.go:25]: can we also make the 2 here a const ? (also at the block sync maybe?)

File: services/gossip/codec/topic_header_sync_test.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+		ResultsBlock:      c.rxContainer,
+	}
+}
+
+
+
+func buildResultsBlockHeaderWithProof(height primitives.BlockHeight) *gossipmessages.ResultsBlockHeaderWithProof {
+	empty32ByteHash := hash.Make32EmptyBytes()
+	rxBlockHeader := (&protocol.ResultsBlockHeaderBuilder{
+		ProtocolVersion:                 primitives.ProtocolVersion(1),
```
Comment (noambergIL) [inline services/gossip/codec/topic_header_sync_test.go:214]: please don't use this PV ... once we have merge with refTime use config.MAXIMAL_PROTOCOL_VERSION_SUPPORTED_VALUE

**Comment by gadcl** [comment]
updated block sync design

**Pull request closed without merging**